vizro-core/tests/unit/vizro/models/_action/test_action.py

```diff
@@ -0,0 +1,205 @@
+    @capture("action")
+    def action_function():
+        return request.param
+
+    return action_function
+
+
+@pytest.fixture
+def callback_context_outputs_grouping(request):
+    """Mock dash.callback_context that represents outputs grouping for custom action."""
+    outputs = request.param
+
+    outputs_grouping = {output: None for output in outputs}
+    mock_callback_context = {"outputs_grouping": {"action_finished": None, **outputs_grouping}}
+    context_value.set(AttributeDict(**mock_callback_context))
+    return context_value
+
+
+class TestActionInstantiation:
+    """Tests model instantiation."""
+
+    def test_create_action_mandatory_only(self, test_action_function):
+        action = Action(function=test_action_function)
+
+        assert hasattr(action, "id")
+        assert action.function == test_action_function
+        assert action.inputs == []
+        assert action.outputs == []
+
+    def test_create_action_mandatory_and_optional(self, test_action_function):
+        inputs = ["component_1.property_A", "component_1.property_B"]
+        outputs = ["component_2.property_A", "component_2.property_B"]
+
+        action = Action(function=test_action_function, inputs=inputs, outputs=outputs)
+
+        assert hasattr(action, "id")
+        assert action.function == test_action_function
+        assert action.inputs == inputs
+        assert action.outputs == outputs
+
+    @pytest.mark.parametrize(
+        "inputs, outputs",
+        [
+            ([], []),
+            (["component.property"], ["component.property"]),
+            (["component.property", "component.property"], ["component.property", "component.property"]),
+            (
+                ["Component_1.Property_A", "Component_2.Property_B"],
+                ["Component_1.Property_A", "Component_2.Property_B"],
+            ),
+        ],
+    )
+    def test_inputs_outputs_valid(self, inputs, outputs, test_action_function):
+        action = Action(function=test_action_function, inputs=inputs, outputs=outputs)
+
+        assert action.inputs == inputs
+        assert action.outputs == outputs
+
+    @pytest.mark.parametrize(
+        "inputs",
+        [
+            None,
+            "string",
+            "component_property",
+            "compo-nent.property",
+            "component.property_1",
+        ],
+    )
+    def test_inputs_invalid(self, inputs, test_action_function):
+        with pytest.raises(Exception):
+            Action(function=test_action_function, inputs=inputs, outputs=[])
+
+    @pytest.mark.parametrize(
+        "outputs",
+        [
+            None,
+            "string",
+            "component_property",
+            "compo-nent.property",
+            "component.property_1",
+        ],
+    )
+    def test_outputs_invalid(self, outputs, test_action_function):
+        with pytest.raises(Exception):
+            Action(function=test_action_function, inputs=[], outputs=outputs)
+
+
+class TestActionBuild:
```

Comment:
Are we only testing this case, as teh creation of the callbacks is hard to test without lower level private functions?